### DISCUSSION TITLE
What is the purpouse of DiracChainEvalSign and why does FCFAconvert fail if it is present?

### DISCUSSION THREAD

**Discussion by SamuelJosephs** [FeynCalc/feyncalc#248] [Q&A] [answered]
I have run into this issue and I am not sure how to proceed, I cannot find where `DiracChainEvalSign` enters into `diracObjectsEval` which triggers the error in the source code. Any pointers would be much appreciated!

For context this is my code with the .gen and .mod files attached:

```
tops = CreateTopologies[0, 1 -> 3]
procs = InsertFields[tops, S[1] -> {F[2, {1}], -F[2, {1}], V[1]}, 
  Model -> "SMEFT_d6_UFO_FULL", GenericModel -> "SMEFT_d6_UFO_FULL"]
Paint[procs]

amp[0] = 
 FCFAConvert[CreateFeynAmp[procs], IncomingMomenta -> {pH}, 
  OutgoingMomenta -> {p1, p2, k}, List -> False, ChangeDimension -> 4,
   TransversePolarizationVectors -> {k}, DropSumOver -> True, 
  SMP -> True, Contract -> True, UndoChiralSplittings -> True]
```

[SMEFT_d6_UFO_FULL.gen.txt](https://github.com/FeynCalc/feyncalc/files/14241307/SMEFT_d6_UFO_FULL.gen.txt)
[SMEFT_d6_UFO_FULL.mod.txt](https://github.com/FeynCalc/feyncalc/files/14241309/SMEFT_d6_UFO_FULL.mod.txt)

And this is the error:
FCFADiracChainJoin has encountered a fatal problem and must abort the \
computation. The problem reads: Evaluation of isolated objects failed.

Any Help is much appreciated!

**Comment by vsht** [comment]
Your model contains undefined symbols carrying Dirac indices and depending on a Dirac matrix:

```
$LoadAddOns = {"FeynArts"};
FAPatch[FAModelsDirectory -> "SMEFT", PatchModelsOnly -> True];
<< FeynCalc`

$FAVerbose = False;
tops = CreateTopologies[0, 1 -> 3];
procs = InsertFields[tops, S[1] -> {F[2, {1}], -F[2, {1}], V[1]}, 
   Model -> 
    FileNameJoin[{NotebookDirectory[], "SMEFT", "SMEFT_d6_UFO_FULL"}],
    GenericModel -> 
    FileNameJoin[{NotebookDirectory[], "SMEFT", "SMEFT_d6_UFO_FULL"}]];
Paint[procs];

amp[0] = 
  FCFAConvert[CreateFeynAmp[DiagramExtract[procs, {6}]], 
   IncomingMomenta -> {pH}, OutgoingMomenta -> {p1, p2, k}, 
   List -> False, ChangeDimension -> 4, 
   TransversePolarizationVectors -> {k}, DropSumOver -> True, 
   SMP -> True, Contract -> True, UndoChiralSplittings -> True, 
   FCFADiracChainJoin -> False];

SelectNotFree[amp[0][[1]], G]
% // InputForm
```

```
G[-1][0][-F[2, {1}], F[Index[Generic, 5]], V[1]][DiracObject[DiracGamma[LorentzIndex[KI1[3]]], DiracGamma[6]][KI1[1], KI1[2]]]*
 (DiracChain[DiracGamma[6], DiracIndex[Dir2], DiracIndex[Dir3]]*G[-1][0][F[2, {1}], -F[Index[Generic, 5]], S[1]][DiracObject[DiracGamma[6]][KI1[1], KI1[2]]] + 
  DiracChain[DiracGamma[6], DiracIndex[Dir3], DiracIndex[Dir2]]*G[-1][0][F[2, {1}], -F[Index[Generic, 5]], S[1]][DiracObject[DiracGamma[6]][KI1[2], KI1[1]]] + 
  DiracChain[DiracGamma[7], DiracIndex[Dir2], DiracIndex[Dir3]]*G[-1][0][F[2, {1}], -F[Index[Generic, 5]], S[1]][DiracObject[DiracGamma[7]][KI1[1], KI1[2]]] + 
  DiracChain[DiracGamma[7], DiracIndex[Dir3], DiracIndex[Dir2]]*G[-1][0][F[2, {1}], -F[Index[Generic, 5]], S[1]][DiracObject[DiracGamma[7]][KI1[2], KI1[1]]])
```

Unless those `G`'s are spelled out explicitly in terms of Dirac matrices, FeynCalc is unable to recover a closed Dirac chain from this kind of input.

**Reply by SamuelJosephs** [reply]
Thank your for your reply, I have had a look at both the .gen and .mod files I linked to in the original post and cannot see these `G[-1][0]`'s anywhere, is it possible FAPatch is introducing them?

I see these `G[+/- 1]`'s in the standard coupling matrices in both my model and Lorentz.gen, they are also a standard part of FeynArts so I'm not sure where they are being introduced into the expressions FeynCalc is processing.

**Reply by SamuelJosephs** [reply]
No I don't think FAPatch is introducing them, but I'm still really not sure how they are ending up in the un `diracChainJoin`ed output.

**Reply by SamuelJosephs** [reply]
As they just aren't in the input .mod/.gen files in a way that isn't in the feynArts specification as detailed [here](https://feynarts.de/FA3Guide.pdf) on page 56

**Reply by SamuelJosephs** [reply]
Ah, it seems that CreateFeynAmp[] is inserting them, curious.

**Reply by SamuelJosephs** [reply, marked as answer]
it was Insertion level, I needed to mandate that `InsertionLevel->{classes}` then it works. Thanks for your help!
